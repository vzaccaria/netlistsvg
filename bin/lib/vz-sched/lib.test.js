const { eventLoop, schedule0 } = require("./lib");

it("Schedules 3 tasks with no waits", () => {
  const res = eventLoop({}, schedule0);
  expect(res).toMatchSnapshot();
});
