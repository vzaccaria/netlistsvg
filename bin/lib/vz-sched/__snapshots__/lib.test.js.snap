// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedules 3 tasks with no waits 1`] = `
Array [
  Object {
    "0": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "0.5": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "1": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "1.5": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "2": Array [
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
    ],
  },
  Object {
    "2.5": Array [
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
    ],
  },
  Object {
    "3": Array [
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
    ],
  },
  Object {
    "3.5": Array [
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
    ],
  },
  Object {
    "4": Array [
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
    ],
  },
  Object {
    "4.5": Array [
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
    ],
  },
  Object {
    "5": Array [
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
    ],
  },
  Object {
    "5.5": Array [
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
    ],
  },
  Object {
    "6": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "6.5": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "7": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "7.5": Array [
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
    ],
  },
  Object {
    "8": Array [
      Object {
        "R": "X",
        "events": Array [
          5.5,
        ],
        "index": 1,
        "lambda": 1,
        "name": "$\\\\tau_2$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 102.5,
      },
      Object {
        "R": "",
        "events": Array [
          6,
        ],
        "index": 2,
        "lambda": 1,
        "name": "$\\\\tau_3$",
        "prev": 0,
        "q": 2,
        "start": 0,
        "sum": 2,
        "vrt": 103,
      },
      Object {
        "R": "",
        "events": Array [
          4.5,
        ],
        "index": 0,
        "lambda": 1,
        "name": "$\\\\tau_1$",
        "prev": 2,
        "q": 2,
        "start": 0,
        "sum": 4,
        "vrt": 104,
      },
    ],
  },
]
`;
